Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -109,7 +109,23 @@ class SourceGenerator {
             return nil
         }
     }
-
+    
+    private func makePlatformFilters(for path: Path, with filters: [SupportedDestination]?, or inferPlatformFiltersByPath: Bool?) -> [String]? {
+        if let filters = filters, !filters.isEmpty {
+            return filters.map { $0.string }
+        } else if inferPlatformFiltersByPath == true {
+            for supportedDestination in SupportedDestination.allCases {
+                let regex1 = try? NSRegularExpression(pattern: "\\/\(supportedDestination)\\/", options: .caseInsensitive)
+                let regex2 = try? NSRegularExpression(pattern: "\\_\(supportedDestination)\\.swift$", options: .caseInsensitive)
```

Comment:
I'm slightly worried about the performance implications of this, but it's opt in

Reply:
I know but this thing is very helpful